### ISSUE TITLE
[BUG] SCSS variables get replaced with environment variables

### ISSUE THREAD

**Issue by jack-michaud** [elkowar/eww#81]
## Describe the bug
It's impossible to use [SCSS variables](https://sass-lang.com/documentation/variables) in `eww.scss`. The way that `eww` parses the SCSS files means that any tokens that look like `$myvariable` get swapped with an environment variable. If the environment variable does not exist, the token gets swapped with a null value, producing invalid SCSS and a null stylesheet.

## Reproducing the issue
Specify a SCSS variable in `eww.scss`. Using a variable name that isn't specified in the environment generates an invalid `scss` file and causes the widget to be unstyled. 

```scss
// eww.scss
window {
  background: #121212;
}

$red: #AE2955;
.red {
  color: $red;
}

scale trough {
  all: unset;
  min-height: 5px;
  border-radius: 50px;
  background-color: darken($red, 20%);
  min-width: 150px;
}
highlight {
  all: unset;
  background-color: red;
  background-color: #ae2955;
  color: rgba(0,0,0,0);
}
slider {
  all: unset;
}
```

```html
<eww>
  <definitions>
    <def name="controls">
      <box class="sidebarContainer controls" orientation="v">
        <box class="red" space-evenly="false" halign="center">
          <box class="icon">
            ☼
          </box>
          <scale min="0" max="100" value="10"></scale>
        </box>
      </box>
    </def>
  </definitions>


  <windows>
    <window name="controls_side">
      <geometry x="50px" y="210px" height="100px" width="250px"/>
      <widget>
        <controls />
      </widget>
    </window>
  </windows>
</eww>

```

## Expected behaviour
Display proper styles while using the specified SCSS variable.

#### Current behavior
![image](https://user-images.githubusercontent.com/13189762/103436606-48b57900-4bd2-11eb-90ca-ee1ec0c08c68.png)
#### Expected behavior
![image](https://user-images.githubusercontent.com/13189762/103436527-60403200-4bd1-11eb-849c-194534329ba5.png)

### Possible solutions

1. Swapping environment variables after the SCSS->CSS conversion
   - Unfortunately this would not work because any undeclared environment variables would throw an error in the SCSS->CSS conversion step
2. Change the environment variable syntax as to not conflict with SCSS (e.g. `${MY_ENV_VAR}`)
   - Would break existing configs that use environment variables, but would prevent a collision with existing SCSS syntax.
3. Leave `$myvar` strings alone when there is not an environment variable to swap it with
   - The current behavior swaps undefined `$variables` with an empty string. It feels like we could happily leave these variables alone and win SCSS variables, but we could run into a situation where there's a name collision with a SCSS variable and we're back at square one; fails to compile when `$CWD: #000` gets turned into `/home/user/: #000`... contrived, but possible. 
4. Intelligently create a list of SCSS variables and don't swap these when we're looking for environment variables
  - [grass's](https://github.com/connorskees/grass) API is not that open, I don't think this is reasonable to do. 

I'm most okay with option 3 if there's a more verbose error message that tells me why my SCSS failed to compile. That allows the user to resolve the issue with their SCSS file.

Any other solutions?

**Comment by elkowar** [comment]
Wow, this was a pretty dumb oversight on my part xD
I'd say the best option by far is option 2, as i feel like trying to work around the ambiguity in any other way would be a pretty bad idea in general (as it would make reading and writing configurations a lot more confusing, and is really not necessary). 
Breaking existing configurations is _not_ a concern at this point, as I've deliberately not yet made an official release of Eww, such as to be able to still break the config format while in the early stage of development.

**Comment by jack-michaud** [comment]
> Wow, this was a pretty dumb oversight on my part xD

Not at all! 

> I'd say the best option by far is option 2

Agreed, this sounds good! I'm glad things are still fluid with config files. I closed WIP PR #82 and I'll whip up a fix with option 2.

**Comment by elkowar** [comment]
I merged #83 , thus this is fixed!
